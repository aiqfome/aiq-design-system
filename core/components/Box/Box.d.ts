import React from 'react';
import { ColorProps, PositionProps, SpaceProps, LayoutProps, FontSizeProps, FontWeightProps, BorderProps, BoxShadowProps, FlexboxProps } from 'styled-system';
export interface Props extends ColorProps, SpaceProps, FlexboxProps, LayoutProps, FontSizeProps, FontWeightProps, PositionProps, BorderProps, BoxShadowProps {
    color?: string;
    children?: any;
    refBox?: any;
    onClick?: any;
    className?: string;
}
export declare const BoxStyled: import("styled-components").StyledComponent<"div", any, {}, never>;
export declare const Box: React.FC<Props>;
//# sourceMappingURL=Box.d.ts.map