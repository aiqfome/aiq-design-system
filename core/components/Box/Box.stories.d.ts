import React from 'react';
declare const _default: {
    component: React.FC<import("./Box").Props>;
    title: string;
};
export default _default;
export declare const basic: React.FC;
//# sourceMappingURL=Box.stories.d.ts.map