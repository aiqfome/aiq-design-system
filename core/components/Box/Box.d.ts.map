{"version":3,"file":"Box.d.ts","sourceRoot":"","sources":["../../../src/components/Box/Box.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAA;AAGzB,OAAO,EAEL,UAAU,EAGV,aAAa,EACb,UAAU,EAEV,WAAW,EAEX,aAAa,EAEb,eAAe,EAEf,WAAW,EAEX,cAAc,EAEd,YAAY,EACb,MAAM,eAAe,CAAA;AAEtB,MAAM,WAAW,KACf,SAAQ,UAAU,EAChB,UAAU,EACV,YAAY,EACZ,WAAW,EACX,aAAa,EACb,eAAe,EACf,aAAa,EACb,WAAW,EACX,cAAc;IAChB,KAAK,CAAC,EAAE,MAAM,CAAA;IACd,QAAQ,CAAC,EAAE,GAAG,CAAA;IACd,MAAM,CAAC,EAAE,GAAG,CAAA;IACZ,OAAO,CAAC,EAAE,GAAG,CAAA;IACb,SAAS,CAAC,EAAE,MAAM,CAAA;CACnB;AAED,eAAO,MAAM,SAAS,oEAUrB,CAAA;AAED,eAAO,MAAM,GAAG,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CAE/B,CAAA"}