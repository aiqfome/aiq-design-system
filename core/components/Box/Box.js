"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Box = exports.BoxStyled = void 0;
const react_1 = __importDefault(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const styled_components_1 = __importDefault(require("styled-components"));
const styled_system_1 = require("styled-system");
exports.BoxStyled = styled_components_1.default.div `
  ${styled_system_1.flexbox}
  ${styled_system_1.color}
  ${styled_system_1.space}
  ${styled_system_1.layout}
  ${styled_system_1.fontSize}
  ${styled_system_1.fontWeight}
  ${styled_system_1.border}
  ${styled_system_1.position}
  ${styled_system_1.boxShadow}
`;
exports.Box = (_a) => {
    var { refBox } = _a, props = __rest(_a, ["refBox"]);
    return react_1.default.createElement(exports.BoxStyled, Object.assign({ "data-testid": 'box', ref: refBox }, props));
};
exports.Box.propTypes = {
    refBox: prop_types_1.default.any
};
