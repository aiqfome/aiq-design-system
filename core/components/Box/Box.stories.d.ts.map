{"version":3,"file":"Box.stories.d.ts","sourceRoot":"","sources":["../../../src/components/Box/Box.stories.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAA;;;;;AAIzB,wBAGC;AAED,eAAO,MAAM,KAAK,EAAE,KAAK,CAAC,EAIzB,CAAA"}