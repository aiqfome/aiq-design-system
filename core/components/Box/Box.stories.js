"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.basic = void 0;
const react_1 = __importDefault(require("react"));
const Box_1 = require("./Box");
exports.default = {
    component: Box_1.Box,
    title: 'Box'
};
exports.basic = () => (react_1.default.createElement(Box_1.Box, { marginBottom: 10, color: 'blue' }, "Design System"));
