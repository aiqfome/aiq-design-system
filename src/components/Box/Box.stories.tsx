import React from 'react'

import { Box } from './Box'

export default {
  component: Box,
  title: 'Box'
}

export const basic: React.FC = () => (
  <Box marginBottom={10} color='blue'>
    Design System
  </Box>
)
