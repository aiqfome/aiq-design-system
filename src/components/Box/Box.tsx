import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import {
  color,
  ColorProps,
  space,
  position,
  PositionProps,
  SpaceProps,
  layout,
  LayoutProps,
  fontSize,
  FontSizeProps,
  fontWeight,
  FontWeightProps,
  border,
  BorderProps,
  boxShadow,
  BoxShadowProps,
  flexbox,
  FlexboxProps
} from 'styled-system'

export interface Props
  extends ColorProps,
    SpaceProps,
    FlexboxProps,
    LayoutProps,
    FontSizeProps,
    FontWeightProps,
    PositionProps,
    BorderProps,
    BoxShadowProps {
  color?: string
  children?: any
  refBox?: any
  className?: string
}

export const BoxStyled = styled.div`
  ${flexbox}
  ${color}
  ${space}
  ${layout}
  ${fontSize}
  ${fontWeight}
  ${border}
  ${position}
  ${boxShadow}
`

export const Box: React.FC<Props> = ({ refBox, ...props }) => {
  return <BoxStyled data-testid='box' ref={refBox} {...props} />
}

Box.propTypes = {
  refBox: PropTypes.any
}
