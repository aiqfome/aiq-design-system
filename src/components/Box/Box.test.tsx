import React from 'react'
import renderer from 'react-test-renderer'

import { Box } from './Box'

describe('must match with the previous snapshot', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Box />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
