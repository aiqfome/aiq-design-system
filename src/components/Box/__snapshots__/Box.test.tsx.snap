// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<div
  className="sc-bdnylx gxDRmI"
/>
`;
